src/pages/workspaces/workspace/Data.js

```diff
@@ -249,9 +259,21 @@ const LocalVariablesContent = ajaxCaller(class LocalVariablesContent extends Com
 })
 
 const ReferenceDataContent = ({ workspace: { workspace: { namespace, attributes } }, referenceKey, loadingWorkspace, firstRender }) => {
-  const selectedData = _.sortBy('key', getReferenceData(attributes)[referenceKey])
+  const [textFilter, setTextFilter] = useState('')
+
+  const selectedData = _.flow(
+    _.filter(data => Utils.textMatch(textFilter, JSON.stringify(data))),
+    _.sortBy('key')
+  )(getReferenceData(attributes)[referenceKey])
   const { initialY } = firstRender ? StateHistory.get() : {}
+
   return h(Fragment, [
+    h(SearchInput, {
+      style: { width: 300, marginBottom: '1rem', borderColor: colors.gray[3], alignSelf: 'flex-end' },
+      placeholder: 'Filter',
+      onChange: ({ target: { value } }) => setTextFilter(value),
```

Comment:
Why does the Ref Data table use hooks and the regular data table uses traditional state for setting the filter?

Reply:
Because the components they're being added into do.